WL-1998

adrianfish opened the pull request:
List size now defaults to 50 and is configurable with citations.default.list.page.size in sakai.properties

adrianfish committed:
WL-1998

List size now defaults to 50 and is configurable with citations.default.list.page.size in sakai.properties

adrianfish committed:
WL-1727

The original scope of allowing html blocks above each citation was abandoned in favour of exposing
the resource description in the list editor.

adrianfish committed:
WL-1763

Notes are now text areas. This requires a clean database or a database conversion.

ColinHebert commented on citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java:
```diff
@@ -463,7 +465,7 @@ public static void clearAdvancedFormState( SessionState state )
 
 	public static final Integer DEFAULT_RESULTS_PAGE_SIZE = new Integer(10);
-	public static final Integer DEFAULT_LIST_PAGE_SIZE = new Integer(10);
+	public static Integer DEFAULT_LIST_PAGE_SIZE = new Integer(50);
```
This one could even be changed into `DEFAULT_LIST_PAGE_SIZE = 50`.

I'm not sure I understand the point of making this a variable, it was (AFAICT) a constant to avoid the magic number (now introduced on L562).

adrianfish replied:
It's not a magic number, it's a variable configurable from sakai.properties. Maybe the name should be lowercased.

ColinHebert replied:
Oh, I see, in that case it should be indeed lowercased. The initialisation could also be removed from here and on L564, instead of logging the absence of `ServerConfigurationService`, manually set `default_list_page_size=50`

ColinHebert commented on citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java:
```diff
@@ -552,6 +554,16 @@ public static void clearAdvancedFormState( SessionState state )
+		ServerConfigurationService scs
+			= (ServerConfigurationService) ComponentManager.get(ServerConfigurationService.class);
+		if(scs != null) {
+			DEFAULT_LIST_PAGE_SIZE = scs.getInt("citations.default.list.page.size", 50);
```
Introduction of a magic number (see L468)

ColinHebert commented on citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java:
```diff
@@ -945,6 +957,13 @@ public String buildListPanelContext(VelocityPortlet portlet, Context context, Ru
+			context.put("description", description);
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
```
Shouldn't this be logged?

adrianfish replied:
You're right. I'll revert the commit on my fork and tidy it up. Can you just ignore the pull request for this particular commit?

ColinHebert commented on citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java:
```diff
@@ -1649,6 +1670,15 @@ public void doFinish ( RunData data)
+	    try {
+			contentService.addProperty(temporaryResourceId,ResourceProperties.PROP_DESCRIPTION,description);
+		} catch (Exception e) {
+			 logger.warn("Failed to set resource description on citation list update. Reason: " + e.getMessage());
```
logging the exception (`logger.warn(String, Throwable)`) would help because the stacktrace could be included in the logs too

ColinHebert commented on citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java:
```diff
@@ -1649,6 +1670,15 @@ public void doFinish ( RunData data)
     	pipe.setRevisedResourceProperty(ResourceProperties.PROP_CONTENT_LENGTH, size_str);
+    	
+	   	String temporaryResourceId = (String) state.getAttribute(CitationHelper.RESOURCE_ID);
+	   	ContentHostingService contentService = (ContentHostingService) ComponentManager.get("org.sakaiproject.content.api.ContentHostingService");
```
using `ComponentManager.get(ContentHostingService.class)` would make refactoring easier.

adrianfish replied:
The rest of the code does it this way though, that's why I did it in this fashion. Lets face it, the entire code base could do with a good refactor. It's arcane.

ColinHebert replied:
I totally agree with that, unfortunately there is a lot of code in the entire project that needs to be refactored/cleaned and this task could take a very long time.
In the mean time, using good practices will help a lot, even if, unfortunately, it introduce another style of code in the project.

adrianfish replied:
If you're happy with me doing another commit on CitationHelperAction I'll replace all those tonight.

adrianfish committed:
Changed the name of the default list size variable to lowercase to reflect its mutabililty.

adrianfish committed:
Revert "Changed the name of the default list size variable to lowercase to reflect its mutabililty."

This reverts commit 42c13bcaede68f480e19b341e6133ff5f478a04e.

adrianfish committed:
WL-1998

Bit of refactoring in accordance with Colins suggestions.

adrianfish committed:
WL-1727

Logged any potential exceptions arising while setting and getting the description.

adrianfish committed:
WL-1875

Sitestats now picks up  reading list reads. You can see if a user has opened a list.

adrianfish committed:
WL-1738

The book chapter and source are now both picked up in the import. On a live database, the sql file, SAK-22496.sql, attached to the ticket needs to be applied.

adrianfish committed:
WL-1732

There's now an extra button which launches the resource picker.

ColinHebert commented on citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java:
```diff
@@ -463,7 +468,8 @@ public static void clearAdvancedFormState( SessionState state )
 	public static final Integer DEFAULT_RESULTS_PAGE_SIZE = new Integer(10);
-	public static final Integer DEFAULT_LIST_PAGE_SIZE = new Integer(10);
+
+	public static Integer defaultListPageSize;
```
Should it be `static` at all? It's set by init.
Do we need to keep it as an `Integer`, it could introduce problems with possible `NullPointerException`.

adrianfish replied:
I can change that as well. Again, just sticking to the established coding convention.

adrianfish committed:
WL-1726

Use jQuery.sortable to reorder citations.

adrianfish committed:
Changed all the string literal class names to .class references. Will make package refactoring easier in future.
